nats-base-client/encoders.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2020 The NATS Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export const TE = new TextEncoder();
```

Comment:
I assume that TextEncoder and decoder don't hold state and that they are "thread safe" meaning that concurrent access to them is ok. I know node is single threaded, but you get what I am saying.

Reply:
correct - they are just functions - also it's single-threaded.